### PR TITLE
Homework 29-04-24

### PR DESCRIPTION
**Pull request by Sargis-Gh**

### CODE REVIEW

File: Saqo/databases/29-04-24/hospital_part_2.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
+select h.address, h.name, d.name, dc.name, p.name from
```
Comment (elya-im) [inline Saqo/databases/29-04-24/hospital_part_2.sql]: use meaningful aliases for the selected columns to make the result readable
e.g 
``` 
select h.address as "Hospital Address"
           ,h.name as "Hospital name"
           ,d.name as "Department name"
           ,dc.name as "Doctor name"
           , p.name as "Patient name" 
```

File: Saqo/databases/29-04-24/hospital_part_2.sql [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+Hospital h inner join Department d on h.id = d.hospital_id
+inner join Doctor dc on dc.id = d.id
+inner join Patient p on p.id = d.hospital_id
+inner join Invoice i on i.patient_id = p.id
+where i.description = 'Medication Cost';
+
+select p.name, p.phone_number from Patient p
+inner join Prescription pr on p.id = pr.patient_id
+where pr.prescription_date like '%-04-%';
```
Comment (elya-im) [inline Saqo/databases/29-04-24/hospital_part_2.sql]: use `MONTH()` function to get the month for the date

File: Saqo/databases/29-04-24/hospital_part_2.sql [hunk 3]
```diff
@@ -0,0 +1,22 @@
+select h.address, h.name, d.name, dc.name, p.name from
+Hospital h inner join Department d on h.id = d.hospital_id
+inner join Doctor dc on dc.id = d.id
+inner join Patient p on p.id = d.hospital_id
+inner join Invoice i on i.patient_id = p.id
+where i.description = 'Medication Cost';
+
+select p.name, p.phone_number from Patient p
```
Comment (elya-im) [inline Saqo/databases/29-04-24/hospital_part_2.sql:12]: use meaningful aliases for the selected columns to make the result readable

File: Saqo/databases/29-04-24/hospital_part_2.sql [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+inner join Prescription pr on p.id = pr.patient_id
+where pr.prescription_date like '%-04-%';
+
+select d.name from Doctor d
```
Comment (elya-im) [inline Saqo/databases/29-04-24/hospital_part_2.sql]: use meaningful aliases for the selected columns to make the result readable

File: Saqo/databases/29-04-24/hospital_part_2.sql [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+select p.name, p.phone_number from Patient p
+inner join Prescription pr on p.id = pr.patient_id
+where pr.prescription_date like '%-04-%';
+
+select d.name from Doctor d 
+inner join Department dep on dep.id = d.department_id
+where dep.name = 'Cardiology';
+
+select dep.name, d.name from Department dep
```
Comment (elya-im) [inline Saqo/databases/29-04-24/hospital_part_2.sql]: use meaningful aliases for the selected columns to make the result readable

File: Saqo/databases/29-04-24/hospital_part_2.sql [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+left join Doctor d on dep.id = d.department_id;
+
+select p.name, p.phone_number, p.address, p.date_of_birth, p.pharmacy_id, ph.name from Patient p
```
Comment (elya-im) [inline Saqo/databases/29-04-24/hospital_part_2.sql]: use meaningful aliases for the selected columns to make the result readable

File: Saqo/databases/29-04-24/hospital_part_2.sql [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+inner join Invoice i on i.patient_id = p.id
+where i.description = 'Medication Cost';
+
+select p.name, p.phone_number from Patient p
+inner join Prescription pr on p.id = pr.patient_id
+where pr.prescription_date like '%-04-%';
+
+select d.name from Doctor d 
+inner join Department dep on dep.id = d.department_id
+where dep.name = 'Cardiology';
```
Comment (elya-im) [inline Saqo/databases/29-04-24/hospital_part_2.sql:18]: there is no need to specify the department, all the doctors should selected with their department name

File: Saqo/databases/29-04-24/hospital_part_2.sql [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+select dep.name, d.name from Department dep
+left join Doctor d on dep.id = d.department_id;
+
+select p.name, p.phone_number, p.address, p.date_of_birth, p.pharmacy_id, ph.name from Patient p
+inner join Pharmacy ph on ph.id = p.pharmacy_id
+inner join Invoice i on i.patient_id = p.id
+where i.cost > 500;
```
Comment (elya-im) [inline Saqo/databases/29-04-24/hospital_part_2.sql:29]: the invoice max value in your data is 100 :rofl:

File: Saqo/databases/29-04-24/hospital_part_2.sql [hunk 9]
```diff
@@ -0,0 +1,22 @@
+select h.address, h.name, d.name, dc.name, p.name from
+Hospital h inner join Department d on h.id = d.hospital_id
+inner join Doctor dc on dc.id = d.id
+inner join Patient p on p.id = d.hospital_id
```
Comment (elya-im) [inline Saqo/databases/29-04-24/hospital_part_2.sql]: `p.id = d.hospital_id` is an incorrect comparison, as they are not the same column

File: Saqo/databases/29-04-24/hospital_part_2.sql (hunk 1 above)
Comment (Sargis-Gh) [reply Saqo/databases/29-04-24/hospital_part_2.sql]: Done ✅

File: Saqo/databases/29-04-24/hospital_part_2.sql (hunk 2 above)
Comment (Sargis-Gh) [reply Saqo/databases/29-04-24/hospital_part_2.sql]: Done ✅

File: Saqo/databases/29-04-24/hospital_part_2.sql (hunk 3 above)
Comment (Sargis-Gh) [reply Saqo/databases/29-04-24/hospital_part_2.sql:12]: Done ✅

File: Saqo/databases/29-04-24/hospital_part_2.sql (hunk 4 above)
Comment (Sargis-Gh) [reply Saqo/databases/29-04-24/hospital_part_2.sql]: Done ✅

File: Saqo/databases/29-04-24/hospital_part_2.sql (hunk 5 above)
Comment (Sargis-Gh) [reply Saqo/databases/29-04-24/hospital_part_2.sql]: Done ✅

File: Saqo/databases/29-04-24/hospital_part_2.sql (hunk 6 above)
Comment (Sargis-Gh) [reply Saqo/databases/29-04-24/hospital_part_2.sql]: Done ✅

File: Saqo/databases/29-04-24/hospital_part_2.sql (hunk 7 above)
Comment (Sargis-Gh) [reply Saqo/databases/29-04-24/hospital_part_2.sql:18]: Done ✅

File: Saqo/databases/29-04-24/hospital_part_2.sql (hunk 8 above)
Comment (Sargis-Gh) [reply Saqo/databases/29-04-24/hospital_part_2.sql:29]: Done ✅

File: Saqo/databases/29-04-24/hospital_part_2.sql (hunk 9 above)
Comment (Sargis-Gh) [reply Saqo/databases/29-04-24/hospital_part_2.sql]: Done ✅

**Review by elya-im — APPROVED** [review]

**Pull request merged**